### PR TITLE
Determine running environment

### PR DESCRIPTION
**Pull request by lucperkins**
This PR ensures that the MNC is less tied to GitHub Actions. I'm not yet sure what will be required to run in GitLab but this at least provides proper hooks and prevents GitHub-specific things from happening there.

### CODE REVIEW

File: magic-nix-cache/src/env.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+use std::{
+    env,
+    fmt::{self, Display},
+};
+
+#[derive(Clone)]
```
Comment (cole-h) [inline magic-nix-cache/src/env.rs:6]:
```suggestion
#[derive(Clone, Copy)]
```

File: magic-nix-cache/src/main.rs [hunk 2]
```diff
@@ -140,6 +159,9 @@ async fn main_cli() -> Result<()> {
     init_logging();
 
     let args = Args::parse();
+    let environment = env::Environment::determine();
+    tracing::debug!("Running in {}", environment.to_string());
+    args.validate(environment.clone())?;
```
Comment (cole-h) [inline magic-nix-cache/src/main.rs:164]:
```suggestion
    args.validate(environment)?;
```

**Comment by cole-h** [comment]
Code looks fine to me. Have you tested this in GHA and GitLab to ensure it does what you expect?

**Comment by lucperkins** [comment]
@cole-h Things seem to be working as normal:

GHA: https://github.com/lucperkins/mnc-action-test/actions/runs/9117773497/job/25069402172
GitLab: https://gitlab.com/DeterminateSystems/flakehub-cache-test/-/jobs/6872721027

**Review by cole-h — APPROVED** [review]

**Pull request merged**